### PR TITLE
add notes on release management to README

### PR DESCRIPTION
**Pull request by boegel**
Hopefully this helps to avoid mistakes like the one fixed in #147

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,3 +98,17 @@ is that it is easy to pull in updates from a feature branch using `git pull`.
...
+When a release of the EESSI test suite is made, the following things must be taken care of:
+
+- Version bump: in both `pyproject.toml` and `setup.cfg`;
+- Release notes: in `RELEASE_NOTES` + in GitHub release (cfr. https://github.com/EESSI/test-suite/releases/tag/v0.2.0);
+- Tag release on GitHub + publish release (incl. release notes);
+- Publishing release to PyPI:
+  ```
+  # example for version 0.2.0
+  python setup.py sdist
+  twine upload dist/eessi_testsuite-0.2.0.tar.gz
```
Comment (casparvl) [inline README.md:113]: Wait... so you can upload new releases to PyPI just like that? I mean, no key/login/identification required?

Comment (boegel) [reply README.md:113]: No, you will need to upload a token, and currently I own the `eessi-testsuite` project on PyPI, so I'm the only one who can publish new releases.
We can look into fixing that when a next release is ready?

**Review by smoors — APPROVED** [review]
lgtm

File: README.md (hunk 1 above)
Comment (casparvl) [reply README.md:113]: Yep, no problem, I just wanted to be sure that the instructions were complete :)

**Pull request merged**